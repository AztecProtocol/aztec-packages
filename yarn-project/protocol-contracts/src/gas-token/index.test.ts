import { GasTokenAddress, getCanonicalGasToken } from './index.js';

describe('GasToken', () => {
  it('returns canonical protocol contract', () => {
    const contract = getCanonicalGasToken();
    expect(contract.address.toString()).toEqual(GasTokenAddress.toString());
  });
});
