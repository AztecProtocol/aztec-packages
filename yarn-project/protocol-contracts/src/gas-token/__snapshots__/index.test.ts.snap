// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GasToken returns canonical protocol contract 1`] = `
{
  "address": AztecAddress {
    "asBigInt": 15665932969535162258465850005057765068325313178683719227533369271008947549337n,
    "asBuffer": {
      "data": [
        34,
        162,
        154,
        149,
        68,
        192,
        203,
        219,
        195,
        91,
        178,
        28,
        170,
        120,
        182,
        240,
        137,
        3,
        190,
        75,
        57,
        10,
        101,
        49,
        158,
        51,
        175,
        215,
        68,
        63,
        52,
        153,
      ],
      "type": "Buffer",
    },
  },
  "instance": {
    "address": AztecAddress {
      "asBigInt": 15665932969535162258465850005057765068325313178683719227533369271008947549337n,
      "asBuffer": {
        "data": [
          34,
          162,
          154,
          149,
          68,
          192,
          203,
          219,
          195,
          91,
          178,
          28,
          170,
          120,
          182,
          240,
          137,
          3,
          190,
          75,
          57,
          10,
          101,
          49,
          158,
          51,
          175,
          215,
          68,
          63,
          52,
          153,
        ],
        "type": "Buffer",
      },
    },
    "contractClassId": Fr {
      "asBigInt": 8048474924228513577213811692516857993138982652659338793381794528774172095058n,
      "asBuffer": {
        "data": [
          17,
          203,
          70,
          126,
          237,
          15,
          145,
          124,
          146,
          221,
          105,
          27,
          232,
          251,
          187,
          118,
          53,
          135,
          128,
          177,
          24,
          140,
          248,
          116,
          36,
          114,
          207,
          163,
          107,
          206,
          178,
          82,
        ],
        "type": "Buffer",
      },
    },
    "initializationHash": Fr {
      "asBigInt": 5411687428194306823799197943510367074755080310726877398642044502900791103391n,
      "asBuffer": {
        "data": [
          11,
          246,
          232,
          18,
          241,
          75,
          176,
          41,
          247,
          203,
          156,
          141,
          168,
          54,
          125,
          217,
          124,
          6,
          142,
          120,
          141,
          79,
          33,
          0,
          127,
          217,
          112,
          20,
          235,
          168,
          207,
          159,
        ],
        "type": "Buffer",
      },
    },
    "portalContractAddress": EthAddress {
      "buffer": {
        "data": [
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
        ],
        "type": "Buffer",
      },
    },
    "publicKeysHash": Fr {
      "asBigInt": 17954372136461894568820199868962143715279932924957662285438325406238856880109n,
      "asBuffer": {
        "data": [
          39,
          177,
          208,
          131,
          154,
          91,
          35,
          186,
          241,
          42,
          141,
          25,
          91,
          24,
          172,
          40,
          143,
          207,
          64,
          26,
          251,
          47,
          112,
          184,
          164,
          181,
          41,
          237,
          229,
          250,
          159,
          237,
        ],
        "type": "Buffer",
      },
    },
    "salt": Fr {
      "asBigInt": 1n,
      "asBuffer": {
        "data": [
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          1,
        ],
        "type": "Buffer",
      },
    },
    "version": 1,
  },
}
`;

exports[`GasToken returns canonical protocol contract 2`] = `
{
  "artifactHash": Fr {
    "asBigInt": 3363571886208995000060653132822906490593381833217164785427349714321114865759n,
    "asBuffer": {
      "data": [
        7,
        111,
        182,
        215,
        73,
        59,
        7,
        90,
        136,
        14,
        238,
        217,
        15,
        236,
        124,
        76,
        1,
        224,
        162,
        77,
        68,
        37,
        34,
        68,
        158,
        77,
        86,
        194,
        99,
        87,
        32,
        95,
      ],
      "type": "Buffer",
    },
  },
  "id": Fr {
    "asBigInt": 8048474924228513577213811692516857993138982652659338793381794528774172095058n,
    "asBuffer": {
      "data": [
        17,
        203,
        70,
        126,
        237,
        15,
        145,
        124,
        146,
        221,
        105,
        27,
        232,
        251,
        187,
        118,
        53,
        135,
        128,
        177,
        24,
        140,
        248,
        116,
        36,
        114,
        207,
        163,
        107,
        206,
        178,
        82,
      ],
      "type": "Buffer",
    },
  },
  "privateFunctions": [
    {
      "isInternal": false,
      "selector": FunctionSelector {
        "value": 2432309179,
      },
      "vkHash": Fr {
        "asBigInt": 0n,
        "asBuffer": {
          "data": [
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
          ],
          "type": "Buffer",
        },
      },
    },
  ],
  "privateFunctionsRoot": Fr {
    "asBigInt": 8909521279374546913798331365196924937704969493731099782202223186665436350504n,
    "asBuffer": {
      "data": [
        19,
        178,
        156,
        63,
        74,
        150,
        235,
        20,
        213,
        211,
        83,
        154,
        99,
        8,
        255,
        151,
        54,
        173,
        93,
        103,
        227,
        246,
        31,
        251,
        183,
        218,
        144,
        142,
        20,
        152,
        8,
        40,
      ],
      "type": "Buffer",
    },
  },
  "publicBytecodeCommitment": Fr {
    "asBigInt": 17070200264896729278819077147914995144552691471904174679230466337970801154331n,
    "asBuffer": {
      "data": [
        37,
        189,
        100,
        22,
        41,
        3,
        133,
        115,
        227,
        6,
        2,
        151,
        250,
        15,
        99,
        123,
        38,
        241,
        249,
        80,
        125,
        120,
        249,
        174,
        143,
        146,
        118,
        52,
        51,
        25,
        25,
        27,
      ],
      "type": "Buffer",
    },
  },
  "version": 1,
}
`;

exports[`GasToken returns canonical protocol contract 3`] = `
[
  {
    "isInternal": false,
    "selector": FunctionSelector {
      "value": 190347739,
    },
  },
  {
    "isInternal": false,
    "selector": FunctionSelector {
      "value": 1776307709,
    },
  },
  {
    "isInternal": false,
    "selector": FunctionSelector {
      "value": 1811750235,
    },
  },
  {
    "isInternal": false,
    "selector": FunctionSelector {
      "value": 3422298534,
    },
  },
]
`;
